zulipterminal/model.py

```diff
@@ -817,6 +817,10 @@ def toggle_stream_pinned_status(self, stream_id: int) -> bool:
     def is_user_subscribed_to_stream(self, stream_id: int) -> bool:
         return stream_id in self.stream_dict
 
+    def is_user_in_unsubscribed_stream(self, stream_id: int) -> bool:
+        return not(self.is_stream_in_list(stream_id, self.pinned_streams)
+                   or self.is_stream_in_list(stream_id, self.unpinned_streams))
```

Comment:
I think we need to refine what `stream_dict` represents, as intuitively I expected these two methods to be the inverse of each other - and so we wouldn't need both.

However, right now I noticed we're not removing entries from stream_dict, so it's going to become a "streams we might be subscribed to" at this point. That doesn't seem overly useful, so I'd suggest we either make it represent "all streams" and check when we look it up in various places that we aren't assuming they're subscribed (through another datastructure), or keep it as subscribed streams only. We would need something like this later for a subscription interface in any case, but for now it should also simplify this case.

Reply:
Would update this shortly.

Reply:
Added another data structure.